src/frequenz/sdk/core/timeseries.py

```diff
@@ -0,0 +1,71 @@
+"""
+Common model definitions used throughout the SDK.
+
+Copyright
+Copyright © 2022 Frequenz Energy-as-a-Service GmbH
+
+License
+MIT
+"""
+from __future__ import annotations
+
+import datetime
+from dataclasses import dataclass
+
+
+@dataclass(frozen=True)
+class TimeSeriesId:
+    """ID for a single metric of a specific component, e.g. 123_active_power."""
+
+    component_id: int
+    metric_id: str
```

Comment:
Once you agree on some name, let me know and I'll change it.

The suggested `time_series_channel_name` could look like this:
```
class TimeSeriesChannelName:
    component_id: int
    metric_id: str

    @property
    def key() -> str
    def from_key(key: str) -> TimeSeriesChannelName

    or

    @property
    def str() -> str
    def from_str(name: str) -> TimeSeriesChannelName

    or

    @property
    def channel_id() -> str
    def from_channel_id(channel_id: str) -> TimeSeriesChannelName
```

in the constructor:
```
self._input_receivers: Dict[TimeSeriesChannelName, Receiver[Optional[Sample]]] = {}
self._output_channels: Dict[TimeSeriesChannelName, Broadcast[Optional[Sample]]] = {}
```

Reply:
> Probably what you are referring to as request_timestamp I call start_time and the class in this PR that would hold (component_id, metric_id, start_time) is Subscription and not TimeSeriesId, but I can change it.

No, we mentioned at some point using `request_timestamp` to sort of deduplicate channels with a start_date that also were requested at the same time. I don't think we'll do it for now. But again, the resampler shouldn't have to care about all these details, it just gets the name of a channel to get data from and the name of a channel to put the data into.

Please don't give a structure to the channel name here, it will be opaque for the resampler.

Reply:
> class TimeSeriesChannelName:
>     component_id: int
>     metric_id: str

Again, the resampler doesn't need to know anything about components, please just use a `str` for identifying a timeseries or channel name.

Reply:
Change the `TimeSeriesId` to a `str` and left `ComponentMetricChannelName` for reference (not just if it will be needed). Let me know if I should delete it.

Reply:
No, it looks like it's coming back, it was a misunderstanding between sahas, tom an me. The resampler will finally be coupled to the data source, and hence to the component_id and metric_id too. But we should probably, maybe, use the same `ComponentMetricRequest` as the data source actor, so let's wait for the PR before touching this again.